Improvement/1481 separate deb rpm packages

ycointe opened the pull request:
**Component**: build, packages, containers, repository

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: Metalk8s over Ubuntu

**Summary**: We need to separate redhat and debian package in the buildchain

**Acceptance criteria**: Redhat and debian package should be managed distinctly in the buildchain.


---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #1481 

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

ycointe committed:
Add constants for RedHat and Debian separation

Add REPO_RPM_ROOT and PKG_RPM_ROOT to the constant list.

ycointe committed:
Change the target for RedHat packages

Change the target for .spec and .meta RedHat packages.

ycointe committed:
Change targets for RedHat repositories

Change the targets for internal repository

ycointe committed:
Split RedHat packages build & download management

Use the new constants, new package and repository target
to build and download packages in a RedHat dedicated folder.

bert-e commented:
# Hello ycointe,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @nootal

* @cmonfort

* @lucieleonard

* @alxf

* @carlito-scality

* @alexandre-allard-scality

* @teddyandrieux

* @slaperche-scality

* @thomasdanan

* @zempashi

* @nicolast

* @sayf-eddine-scality

* @monpote

* @gdemonet

* @tcarmet

slaperche-scality commented on buildchain/buildchain/constants.py:
```diff
@@ -29,10 +29,14 @@
 ISO_IMAGE_ROOT : Path = ISO_ROOT/'images'
 # Root for the packages that we build ourselves.
 PKG_ROOT : Path = config.BUILD_ROOT/'packages'
+#Root for the RPM package that we build ourselves.
```
minor: Missing space after `#`

slaperche-scality commented on buildchain/buildchain/targets/package.py:
```diff
@@ -106,12 +106,13 @@ def srcdir(self) -> Path:
     def meta(self) -> Path:
         """.meta file path."""
-        return self.rootdir/'{}.meta'.format(self.name)
+        #return self.rootdir/'redhat'/'{}.meta'.format(self.name)
```
the comment can be removed I guess

slaperche-scality replied:
This is wrong, as we have seen.

Instead of fixing it in a later commit, I would squash the commit `Change repositories target ` into this one.

slaperche-scality commented on buildchain/buildchain/packaging.py:
```diff
@@ -66,19 +66,31 @@ def task__package_mkdir_root() -> types.TaskDict:
     ).task
 
+def task__package_mkdir_rpm_root() -> types.TaskDict:
+    """Create the redhat packages root directory."""
```
s/redhat/RedHat/, same in `task__package_mkdir_rpm_iso_root` docstring

slaperche-scality commented on buildchain/buildchain/constants.py:
```diff
@@ -29,10 +29,14 @@
 ISO_ROOT : Path = config.BUILD_ROOT/'root'
 # Root of the repositories on the ISO.
 REPO_ROOT : Path = ISO_ROOT/'packages'
+# Root of the redhat repositories on the ISO.
```
s/redhat/RedHat/ in the comment.

ycointe committed:
Change repositories target

Change the path where the root packages are and
where the srpm are put.

ycointe committed:
Change docker mount point

Mount point changed to download packages to the
right place.

ycointe committed:
Change some paths used by Vagrant

Change the relative directory and the repository
used by vagrant.

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/1481-separate-deb-rpm-packages` and the
destination branch `development/2.4`.

Please resolve the conflict on **the feature branch** (`improvement/1481-separate-deb-rpm-packages`).

```bash
 $ git fetch
 $ git checkout origin/improvement/1481-separate-deb-rpm-packages
 $ git merge origin/development/2.4
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:improvement/1481-separate-deb-rpm-packages
```

slaperche-scality commented on buildchain/buildchain/constants.py:
```diff
@@ -29,10 +29,14 @@
 ISO_IMAGE_ROOT : Path = ISO_ROOT/'images'
 # Root for the packages that we build ourselves.
 PKG_ROOT : Path = config.BUILD_ROOT/'packages'
+# Root for the RPM package that we build ourselves.
```
package**s**

slaperche-scality commented on buildchain/buildchain/packaging.py:
```diff
@@ -66,19 +66,31 @@ def task__package_mkdir_root() -> types.TaskDict:
         directory=constants.PKG_ROOT, task_dep=['_build_root']
     ).task
 
+def task__package_mkdir_rpm_root() -> types.TaskDict:
```
In commit message:
- s/Constant variables/constants/
- s/redhat/RedHat/

slaperche-scality commented on buildchain/buildchain/docker_command.py:
```diff
@@ -155,11 +155,11 @@ class DockerRun:
 
     RPMLINTRC_MOUNT : Mount = bind_ro_mount(
```
commit message: s/good/right/

slaperche-scality commented on salt/metalk8s/defaults.yaml:
```diff
@@ -36,7 +36,7 @@ repo:
     common_registry: '99-registry-common.inc'
   local_mode: false
-  relative_path: packages     # relative to ISO root (configured in pillar)
+  relative_path: packages/redhat     # relative to ISO root (configured in pillar)
```
commit message: s/path tu use/paths used by/

slaperche-scality reviewed (commented):
LGTM, just need to fix some typo in commit messages and squash one commit.

(not approving yet because you need to solve some conflicts I think)

jbertran commented on buildchain/buildchain/packaging.py:
```diff
@@ -87,15 +99,15 @@ def clean() -> None:
             coreutils.rm_rf(repository.rootdir)
 
-    pkg_list = constants.ROOT/'packages/packages.list'
+    pkg_list = constants.ROOT/'packages/redhat/packages.list'
```
Minor: these probably should have used pathlib from the start.

```diff
- constants.ROOT/'packages/redhat/packages.list'
+ constants.ROOT/'packages'/'redhat'/'packages.list'
```

jbertran commented on buildchain/buildchain/packaging.py:
```diff
@@ -136,13 +148,13 @@ def task__build_repositories() -> Iterator[types.TaskDict]:
     name='metalk8s-build',
     version='latest',
-    dockerfile=constants.ROOT/'packages'/'Dockerfile',
+    dockerfile=constants.ROOT/'packages/redhat/Dockerfile',
```
You dropped pathlib `/` usage here, should look similar to above comment.

jbertran commented on buildchain/buildchain/packaging.py:
```diff
@@ -136,13 +148,13 @@ def task__build_repositories() -> Iterator[types.TaskDict]:
     file_dep=[
-        constants.ROOT/'packages/yum_repositories/kubernetes.repo',
-        constants.ROOT/'packages/yum_repositories/saltstack.repo'
+        constants.ROOT/'packages/redhat/yum_repositories/kubernetes.repo',
```
Ditto. In fact, most of these could make use of `constants.PKG_RPM_ROOT`

ycointe replied:
This is not the good constant. this one is `_build/packages/redhat`.

jbertran replied:
Oops, Indeed. For less confusion, we should probably have a constant for `_build/packages`, something like `PKG_STAGING`.

That can be done in another PR as a drive-in change though, let's get this one in first.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @nootal

* @cmonfort

* @lucieleonard

* @alxf

* @carlito-scality

* @alexandre-allard-scality

* @teddyandrieux

* @slaperche-scality

* @thomasdanan

* @zempashi

* @nicolast

* @sayf-eddine-scality

* @monpote

* @gdemonet

* @tcarmet

jbertran reviewed (commented):
Nice work, just a couple of minor comments for pathlib

slaperche-scality commented on ui/package.json:
```diff
@@ -5,7 +5,7 @@
   "dependencies": {
     "@fortawesome/fontawesome-free": "^5.7.2",
     "@kubernetes/client-node": "github:scality/kubernetes-client-javascript.git#browser-dist",
-    "@scality/core-ui": "github:scality/core-ui.git#improvement/add-className-in-input-error",
```
How does this belong to your PR? oO

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @jbertran

* @tcarmet

* @carlito-scality

* @slaperche-scality

* @alexandre-allard-scality

* @zempashi

* @teddyandrieux

* @sayf-eddine-scality

* @nootal

* @nicolast

* @lucieleonard

* @gdemonet

* @monpote

* @thomasdanan

ycointe committed:
Move all required files for packages installation

All these files have been moved in a directory "redhat"
Another one, named "debian" has also been created to handle
Debian packages installation.

It will permit the separation between RedHat and Debian packages
management.

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/1481-separate-deb-rpm-packages` and the
destination branch `development/2.4`.

Please resolve the conflict on **the feature branch** (`improvement/1481-separate-deb-rpm-packages`).

```bash
 $ git fetch
 $ git checkout origin/improvement/1481-separate-deb-rpm-packages
 $ git merge origin/development/2.4
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:improvement/1481-separate-deb-rpm-packages
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @jbertran

* @tcarmet

* @carlito-scality

* @slaperche-scality

* @alexandre-allard-scality

* @zempashi

* @teddyandrieux

* @sayf-eddine-scality

* @nootal

* @nicolast

* @lucieleonard

* @gdemonet

* @monpote

* @thomasdanan

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @gdemonet

* @monpote

* @sayf-eddine-scality

* @alexandre-allard-scality

* @zempashi

* @carlito-scality

* @lucieleonard

* @thomasdanan

* @teddyandrieux

* @nootal

* @cmonfort

* @slaperche-scality

* @nicolast

* @jbertran

ycointe commented:
@bert-e approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @gdemonet

* @monpote

* @sayf-eddine-scality

* @alexandre-allard-scality

* @zempashi

* @carlito-scality

* @lucieleonard

* @thomasdanan

* @teddyandrieux

* @nootal

* @cmonfort

* @slaperche-scality

* @nicolast

* @jbertran










*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.4`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.4`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



Please check the status of the associated issue None.

Goodbye ycointe.
